static/app/components/compactSelect/listBox/index.tsx

```diff
@@ -131,16 +131,13 @@ const ListBox = forwardRef<HTMLUListElement, ListBoxProps>(function ListBox(
     ref
   );
 
-  const onKeyDown = useCallback<React.KeyboardEventHandler<HTMLUListElement>>(
-    e => {
-      const continueCallback = keyDownHandler?.(e);
-      // Prevent list box from clearing value on Escape key press
-      if (continueCallback && e.key !== 'Escape') {
-        listBoxProps.onKeyDown?.(e);
-      }
-    },
-    [keyDownHandler, listBoxProps]
-  );
+  const onKeyDown = (e: React.KeyboardEvent<HTMLUListElement>) => {
```

Comment:
Were the useCallbacks removed here because the effects they were observing not stable?